### PR TITLE
Add type effect (#804)

### LINKED ISSUE #804: Implement typewritting effect on RelevantWord component
Following comment at https://github.com/Qiskit/qiskit.org/pull/802#discussion_r463436947, this task is about to rename the component and encapsulate a typewritting effect:

Typewriter effect

### PR DESCRIPTION
**Pull request by techtolentino**
Work in Progress

**Issue**
Resolves #804

**Description**
This PR adds a `TypedWord` component that will replace the `RelevantWord` component, on the Learn Page

### CODE REVIEW

File: components/ui/TypedWord.vue [hunk 1]
```diff
@@ -0,0 +1,91 @@
+<template>
+  <div class="typed-text__container" :class="{ 'typed-text__cursor': showCursor }">
```
Comment (delapuente) [inline components/ui/TypedWord.vue:2]: We follow a convention in which the main container has a class named after the name of the component in kebab-case so please, could you name this `div` class `typed-word`?

File: components/ui/TypedWord.vue [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    <span class="typed-text__content">{{typeValue}}</span>
+  </div>
+</template>
+
+<script>
+export default {
```
Comment (delapuente) [inline components/ui/TypedWord.vue:8]: In qiskit.org we follow TypeScript class-based components as shown in [Nuxt.js documentation](https://typescript.nuxtjs.org/cookbook/components/#template) (see "Class API" tab). Could you rewrite this component to follow this style, please?

File: components/ui/TypedWord.vue [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      typeValue: "",
+      typeStatus: false,
+      typeArray: ["Qiskit", "Python", "Quantum"],
+      typingSpeed: 200,
+      erasingSpeed: 100,
+      newTextDelay: 3000,
+      eraseTextDelay: 5000,
+      typeArrayIndex: 0,
+      charIndex: 0,
+      showCursor: true,
```
Comment (delapuente) [inline components/ui/TypedWord.vue:20]: We should identify the public API of the component (those values that customize the behaviour of the component) and convert them into component properties so it is the consumer of the component who's in charge of customizing the behaviour to their needs.

The way it is right now implies a strong coupling with the learn page.

**Review by delapuente — CHANGES_REQUESTED** [review]
This is looking nice @techtolentino. I would recommend starting with a word already written so it does not seem like broken on first load.

Please, review the [build logs](https://travis-ci.com/github/Qiskit/qiskit.org/builds/184232066) and solve the issues regarding the linter.

File: components/learn/TheLearnHeader.vue [hunk 4]
```diff
@@ -1,16 +1,20 @@
 <template>
   <ThePageHeader class="the-learn-header">
-    Start your path towards<br>
-    learning <RelevantWord>Qiskit</RelevantWord>
+    Start your path towards
+    <br />learning
+    <TypedWord
+      :initialWord="'Qiskit'"
```
Comment (delapuente) [inline components/learn/TheLearnHeader.vue:6]: Use kebab-case here, it is automatically converted into camelCase by Nuxt.js.

Comment (delapuente) [reply components/learn/TheLearnHeader.vue:6]: Also, you don't need this to be dynamic so it is better to use:
```suggestion
      :initial-word="Qiskit"
```

File: components/ui/TypedWord.vue [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+<style lang="scss" scoped>
+.typed-word {
+  &__content {
+    font-style: italic;
+    color: white;
+    padding: $spacing-01 $spacing-03 $spacing-01 0;
+    margin-left: $spacing-02;
+  }
+  &__container {
```
Comment (delapuente) [inline components/ui/TypedWord.vue:90]: Add blank lines between selectors.

File: components/ui/TypedWord.vue [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+</template>
+
+<script>
+// Implementation based on https://github.com/webnoobcodes/vuejs-typeeffect
+import Vue from 'vue'
+import { Component, Prop } from 'vue-property-decorator'
+
+@Component
+export default class extends Vue {
+  @Prop({ type: String, default: '' }) initialWord
```
Comment (delapuente) [inline components/ui/TypedWord.vue:14]: Add TypeScript annotations such as:
```suggestion
  @Prop({ type: String, default: '' }) initialWord: string = ''
```

File: components/ui/TypedWord.vue [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+  @Prop({ type: Array, default: [] }) typeArray
+  @Prop({ type: Number, default: 200 }) typingSpeed
+  @Prop({ type: Number, default: 100 }) erasingSpeed
+  @Prop({ type: Number, default: 3000 }) eraseTextDelay
+  @Prop({ type: Number, default: 3000 }) newTextDelay
+
+  // Initial data
+  value = ''
```
Comment (delapuente) [inline components/ui/TypedWord.vue:22]: Same here:
```suggestion
  value: string = ''
```
We try to leverage TypeScript type inference although it is considered a good practice to annotate declarations and signatures.

File: components/ui/TypedWord.vue [hunk 8]
```diff
@@ -0,0 +1,113 @@
+<template>
+  <div class="typed-word__container" :class="{ 'typed-word__cursor': showCursor }">
```
Comment (delapuente) [inline components/ui/TypedWord.vue:2]: The top-level element should include a BEM block class named after the component so this element should include the class `typed-word`. I argue that there is no need for adding the container too although you can try to convince me. ;)

Comment (delapuente) [reply components/ui/TypedWord.vue:2]: Correct me if I'm wrong, please: the class `typed-word__cursor` is there for signalling the cursor is visible, is it right? If that's the case, consider converting it into a [boolean BEM block modifier](https://en.bem.info/methodology/quick-start/#boolean) like `typed-word_cursor-visible`.

File: components/ui/TypedWord.vue [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+  }
+  &__container {
+    background-color: $purple-70;
+    display: inline-block;
+  }
+
+  &__cursor {
+    animation: 0.75s flash step-end infinite;
+
+    & .typed-word__content {
```
Comment (delapuente) [inline components/ui/TypedWord.vue:98]: You don't need to prepend `&` for making a descendant selector. It is enough with:
```suggestion
    .typed-word__content {
```

**Review by delapuente — CHANGES_REQUESTED** [review]
Getting there! Thank you @techtolentino! Please, implement the changes I ask for here.

File: components/ui/TypedWord.vue (hunk 6 above)
Comment (techtolentino) [reply components/ui/TypedWord.vue:14]: @delapuente - I'm running into another prop mutation issue, when adding the Ts annotation here.

Screen Shot 2020-09-17 at 7 19 16 AM


and the only place it's being referred to is here:

```js
created () {
  this.value = this.initialWord
  setTimeout(this.eraseText, this.newTextDelay + 200)
}
```

any thoughts?

Comment (techtolentino) [reply components/ui/TypedWord.vue]: (actually, I think I fixed it)

**Pull request merged**